### DISCUSSION TITLE
list of supported video format?

### DISCUSSION THREAD

**Discussion by Mikii67** [SafeExamBrowser/seb-win-refactoring#745] [Q&A]
I understand that, in order for an external video to be visible in SEB and on different platforms, this needs to be encoded using an "open source codec" (excluding mp4). 
Can someone **post a list of video and audio codecs** that will work with SEB both on windows and macos?

I am struggling to find a suitable "universal" solution here.

Thank you.

**Comment by dbuechel** [comment]
You can open e.g. https://html5test.com/ in SEB for Windows and macOS, there you'll find which video and audio codecs are supported by which version respectively.

**Reply by Mikii67** [reply]
Thanks for this, I did not know this tool. I'll give it a try.

**Comment by danschlet** [comment]
Apparently Apple added support for WebM in macOS 11.3. As SEB for macOS always uses the WebKit browser engine of the macOS version it runs on, it should theoretically also work from macOS 11.3, but you would need to test it (and make sure SEB settings allow the modern WebView to be used, see macOS release notes). 

Some sources claim that iOS 15 should also support WebM, but I wasn't able to verify that, also HTML5 Test doesn't show support for it on an iPad with iPadOS 16 or 17.

If you want video on all platforms and older macOS versions, you would need to provide both WebM and MP4 videos and use a Javascript video player on your website which chooses the right format for the accessing browser.

**Reply by Mikii67** [reply]
I did not try WebM yet, I'll do it soon. Thanks.
